### PR TITLE
refactor: use deno & emit and remove vite

### PR DESCRIPTION
**Pull request by EdJoPaTo**
Move some files, everything in public is now the public folder to be deployed.
Also run `demo fmt` to use a common code style.

### CODE REVIEW

**Comment by yuki-93** [comment]
What are denos presets for ts? I want to have "noExplicitAny" and "strict" to enforce types and avoid "anys". Is this the case?

Can you explain me, how the automation with the release will work or what I have to to? 

Thanks for your work so far. 

Also will test this on my machine later, so takes some time to apply this PR.

**Comment by EdJoPaTo** [comment]
> What are denos presets for ts? I want to have "noExplicitAny" and "strict" to enforce types and avoid "anys". Is this the case?

Both you mentioned are [enabled by default](https://deno.land/manual/advanced/typescript/configuration).

[Personally I consider to enable even more tsconfig rules to be even more strict](https://github.com/EdJoPaTo/deno-runtime-template/blob/6dde7104c24d7af5cf76b029340a21814509d343/deno.jsonc#L7-L16) but this would create more churn in this PQ which is why I didn't.

> Can you explain me, how the automation with the release will work or what I have to to?

On every commit the CI is run to do stuff like `demo lint`, `deno check *.ts` and `deno task bundle`. The last one is added as a build artefact so you can simply take it from every run and deploy the given build with a static web server.

On tags starting with v a GitHub release is generated and the artefact added to it.

**Comment by yuki-93** [comment]
Feel free to create an issue for the following ts config work ☺️

**Comment by yuki-93** [comment]
> Feel free to create an issue for the following ts config work ☺️

Created #8 for the following ts rules

**Review by yuki-93 — CHANGES_REQUESTED** [review]

File: deno.jsonc [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+  // https://deno.land/manual/getting_started/configuration_file
+  "lock": false,
+  "tasks": {
+    "bundle": "deno run --allow-env --allow-read --allow-write=public/main.js --allow-net=deno.land bundle.ts"
```
[the remark this answers is not in the document]
Comment (EdJoPaTo) [reply deno.jsonc:5]: What do you want as "dev-mode"?

[the remark this answers is not in the document]
Comment (yuki-93) [reply deno.jsonc:5]: Some kind of dev-server for local development with hot module reloading :) 
So start a dev command in the sh and let it run, changes are automatically applied.

File: bundle.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { bundle } from "https://deno.land/x/emit@0.27.0/mod.ts";
+const { code } = await bundle("main.ts", {
+  compilerOptions: {
+    inlineSourceMap: true,
+    inlineSources: true,
+  },
+});
+await Deno.writeTextFile("public/main.js", code);
```
[the remark this answers is not in the document]
Comment (EdJoPaTo) [reply bundle.ts:8]: As this is static content a server like nginx will generate [ETags](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag) to handle reduce traffic.
Changing the filename on different builds requires the HTML to have a short caching time while the JS/CSS would need high caching times to be useful.
Currently the production web-server uses only ETags and no caching time at all making this approach basically useless. Personally I think it is overkill for such a small project which is rarely updated anyway.

File: main.ts [hunk 3]
```diff
@@ -1,21 +1,20 @@
-import './style.css'
+const getRandomItem = (items: Array<string>): string =>
+  items[Math.floor(Math.random() * items.length)];
 
-const getRandomItem = (items: Array<string>): string => items[Math.floor(Math.random()*items.length)]
-
-const prepareData = async (): Promise<Array<Array<string>>> => {
+async function prepareData(): Promise<Array<Array<string>>> {
```
[the remark this answers is not in the document]
Comment (EdJoPaTo) [reply main.ts:4]: Personally I think a function stated as a function is way more readable but its your project.

**Review by yuki-93 — CHANGES_REQUESTED** [review]
Looks fine, waiting for dev mode to be added.

File: deno.jsonc (hunk 1 above)
[the remark this answers is not in the document]
Comment (EdJoPaTo) [reply deno.jsonc]: 599035b3fc3a88952aeaf584ac63fd0cc6c6e8c8

**Review by yuki-93 — COMMENTED** [review]
Thanks🤗. Will test this later on, but Code looks fine so far.

**Comment by yuki-93** [comment]
deno seem to be not the right tool for this. So closing this PR. But thanks for your work and contribution.

**Pull request closed without merging**